index.bs

```diff
@@ -581,6 +595,26 @@ To <dfn>handle an incoming message</dfn> given a [=WebSocket connection=]
 
 </div>
 
+<div algorithm>
+To <dfn export>emit an event</dfn> given |body| and |related contexts|:
+
+ 1. [=Assert=]: |body| has size 2 and contains "<code>method</code>" and "<code>params</code>".
+
+ 1. If the [=current session=] is null, or the [=current session=]'s [=WebSocket
+    Connection=] is null then return.
+
+ 1. If [=event is enabled=] given [=current session=],
+    |body|["<code>method</code>"] and |related contexts|:
```

Comment:
It's a bit unfortunate that "emit an event" needs to take a map with exactly two things, instead of event type and params separately. At the call sites, "emit an event named 'foo.bar' with *params*" would be nicer, and one wouldn't need to peek inside the map here.

The reason for this is basically how the CDDL is written, pairing together event name and the parameters in a way where they can't be mismatched, which is also nice.

I don't have a suggested solution here, but raising it in case someone can see a way to make it nice also at the call sites.